File: archive/scripts/check-schema.js
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config();

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseServiceKey) {
  console.error('Missing Supabase environment variables');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseServiceKey);

async function checkSchema() {
  console.log('Checking current database schema...\n');
  
  // Check vulnerabilities table
  const { data: vulnData, error: vulnError } = await supabase
    .from('vulnerabilities')
    .select('*')
    .limit(1);
  
  if (!vulnError && vulnData && vulnData.length > 0) {
    console.log('Vulnerabilities table fields:');
    console.log(Object.keys(vulnData[0]));
    console.log('Sample record:', JSON.stringify(vulnData[0], null, 2));
  } else {
    console.log('Vulnerabilities table error:', vulnError);
  }
  
  // Check options_for_consideration table
  const { data: ofcData, error: ofcError } = await supabase
    .from('options_for_consideration')
    .select('*')
    .limit(1);
  
  if (!ofcError && ofcData && ofcData.length > 0) {
    console.log('\nOptions for Consideration table fields:');
    console.log(Object.keys(ofcData[0]));
    console.log('Sample record:', JSON.stringify(ofcData[0], null, 2));
  } else {
    console.log('Options for Consideration table error:', ofcError);
  }
  
  // Check submissions table
  const { data: subData, error: subError } = await supabase
    .from('submissions')
    .select('*')
    .limit(1);
  
  if (!subError && subData && subData.length > 0) {
    console.log('\nSubmissions table fields:');
    console.log(Object.keys(subData[0]));
    console.log('Sample record:', JSON.stringify(subData[0], null, 2));
  } else {
    console.log('Submissions table error:', subError);
  }
}

checkSchema();

File: vofc-viewer/archive/scripts/check_schema.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

# Load credentials
load_dotenv("../../.env")
SUPABASE_URL = os.getenv("SUPABASE_URL") or "https://wivohgbuuwxoyfyzntsd.supabase.co"
SUPABASE_KEY = os.getenv("SUPABASE_SERVICE_ROLE_KEY") or "sb_secret_o6CXg_vuAGMAvnnsfgNDqw_9ZQoQyUk"

supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)

print("CHECKING DATABASE SCHEMA")
print("=" * 50)

# Check sources table
print("Sources table:")
sources = supabase.table("sources").select("*").limit(1).execute()
if sources.data:
    print("Columns:", list(sources.data[0].keys()))
else:
    print("No data in sources table")

# Check vulnerabilities table
print("\nVulnerabilities table:")
vulns = supabase.table("vulnerabilities").select("*").limit(1).execute()
if vulns.data:
    print("Columns:", list(vulns.data[0].keys()))
else:
    print("No data in vulnerabilities table")

# Check OFCs table
print("\nOptions for Consideration table:")
ofcs = supabase.table("options_for_consideration").select("*").limit(1).execute()
if ofcs.data:
    print("Columns:", list(ofcs.data[0].keys()))
else:
    print("No data in options_for_consideration table")
